hooks/useSubmission.tsx

```diff
@@ -408,130 +440,173 @@ function SubmissionControllerCreator({
     throw new Error("SubmissionContext not found");
   }
   const submissionController = ctx.submissionController;
-  const { query } = useShow<SubmissionWithFilesGraderResultsOutputTestsAndRubric>({
+
+  // Register all generic data types BEFORE setting up live subscriptions
+  submissionController.registerGenericDataType(
+    "submission_file_comments",
+    (item: unknown) => (item as SubmissionFileComment).id
+  );
+  submissionController.registerGenericDataType(
+    "submission_comments",
+    (item: unknown) => (item as SubmissionComments).id
+  );
+  submissionController.registerGenericDataType(
+    "submission_reviews",
+    (item: unknown) => (item as SubmissionReviewWithRubric).id
+  );
+  submissionController.registerGenericDataType(
+    "submission_artifact_comments",
+    (item: unknown) => (item as SubmissionArtifactComment).id
+  );
+
+  // Single comprehensive query to load all data upfront
+  const { query } = useShow<SubmissionWithAllRelatedData>({
     resource: "submissions",
     id: submission_id,
     meta: {
-      select:
-        "*, assignments(*, rubrics!grading_rubric_id(*,rubric_criteria(*,rubric_checks(*)))), submission_files(*), assignment_groups(*, assignment_groups_members(*, profiles!profile_id(*))), grader_results(*, grader_result_tests(*), grader_result_output(*)), submission_artifacts(*)"
+      select: `
+        *,
+        assignments(*, rubrics!grading_rubric_id(*,rubric_criteria(*,rubric_checks(*)))),
+        submission_files(*),
+        assignment_groups(*, assignment_groups_members(*, profiles!profile_id(*))),
+        grader_results(*, grader_result_tests(*), grader_result_output(*)),
+        submission_artifacts(*),
+        submission_file_comments(*),
+        submission_comments(*),
+        submission_reviews!submission_reviews_submission_id_fkey(*, rubrics(*, rubric_criteria(*, rubric_checks(*)))),
+        submission_artifact_comments!submission_artifact_comments_submission_id_fkey(*)
+      `.trim()
     }
   });
-  const { data: liveFileComments, isLoading: liveFileCommentsLoading } = useList<SubmissionFileComment>({
+
+  // Set up live subscriptions with proper event handling
+  // We need these enabled to receive live events, but we'll ignore the initial data since we already loaded it
+  const [liveSubscriptionsReady, setLiveSubscriptionsReady] = useState(false);
+
+  useList<SubmissionFileComment>({
     resource: "submission_file_comments",
     filters: [{ field: "submission_id", operator: "eq", value: submission_id }],
     liveMode: "manual",
-    pagination: {
-      pageSize: 1000
```

Comment:
Without pageSize=1,000, this will need to handle pagination to fetch all pages, right?